src/python/pants/pantsd/service/scheduler_service.py

```diff
@@ -96,7 +155,7 @@ def _process_event_queue(self):
     try:
       subscription, is_initial_event, files = (event['subscription'],
                                                event['is_fresh_instance'],
-                                               [f.decode('utf-8') for f in event['files']])
+                                               {f.decode('utf-8') for f in event['files']})
```

Comment:
I'm doing an `in` check, which feels more set-y than list-y, but happy to change back to a list if there's a good reason to avoid a set

Reply:
since invalidation vs pidfile checking are keyed into two different named watchman subscriptions, how would you feel about just checking the value of `subscription` here and directly dispatching to the pidfile handler on those events vs doing the `in` check for all files for all subscriptions?